File: backend/app/schemas/service.py
from pydantic import BaseModel
from datetime import datetime


class ServiceBase(BaseModel):
    ServiceName: str
    Description: str
    Icon: str
    Status: str


class ServiceCreate(ServiceBase):
    pass


class ServiceUpdate(ServiceBase):
    pass


class ServiceResponse(ServiceBase):
    ServiceID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True

File: frontend/types/service.ts
export interface Service {
    ServiceID: number;
    ServiceName: string;
    Description: string;
    Icon: string;
    Status: string;
    CreatedAt: string;
  }
  
  export interface ServiceCreate {
    ServiceName: string;
    Description: string;
    Icon: string;
    Status: string;
  }
  
  export interface ServiceUpdate {
    ServiceName: string;
    Description: string;
    Icon: string;
    Status: string;
  }